Core/AppRuntime/V8Inspector/Source/V8InspectorSocket.cpp

```diff
@@ -642,7 +639,7 @@ void TcpHolder::SetHandler(ProtocolHandler* handler) {
   handler_ = handler;
 }
 
-int TcpHolder::WriteRaw(const std::vector<char>& buffer/*, uv_write_cb write_cb*/) {
```

Comment:
Just for my understanding - why can't we pass buffer in as a const reference?

Reply:
This is an optimization. Passing as const reference will eventually cause the code to copy the buffer. Either we use const reference from beginning to end or use move semantics all the way through. Both will work but a mixture will cause an additional copy for no reason.

Reply:
Actually, I'm wrong about both will work. If we use const references everywhere, it will still result in an unnecessary copy when we need to use it with `asio`.